Source/stores.cpp

```diff
@@ -204,152 +209,150 @@ void StoreAutoPlace()
 	}
 }
 
-void S_StartSmith()
-{
-	stextsize = false;
-	stextscrl = false;
-	AddSText(0, 1, true, "Welcome to the", COL_GOLD, false);
-	AddSText(0, 3, true, "Blacksmith's shop", COL_GOLD, false);
-	AddSText(0, 7, true, "Would you like to:", COL_GOLD, false);
-	AddSText(0, 10, true, "Talk to Griswold", COL_BLUE, true);
-	AddSText(0, 12, true, "Buy basic items", COL_WHITE, true);
-	AddSText(0, 14, true, "Buy premium items", COL_WHITE, true);
-	AddSText(0, 16, true, "Sell items", COL_WHITE, true);
-	AddSText(0, 18, true, "Repair items", COL_WHITE, true);
-	AddSText(0, 20, true, "Leave the shop", COL_WHITE, true);
-	AddSLine(5);
-	storenumh = 20;
-}
-
-void S_ScrollSBuy(int idx)
+void S_Scroll(int idx)
 {
-	int l, ls;
-
-	ls = idx;
 	ClearSText(5, 21);
 	stextup = 5;
 
-	for (l = 5; l < 20; l += 4) {
-		if (!smithitem[ls].isEmpty()) {
+	for (int l = 5; l < 20; l += 4) {
+		if (idx >= storenumh)
+			break;
+		if (!storeItem[idx].isEmpty()) {
 			text_color iclr = COL_WHITE;
-			if (smithitem[ls]._iMagical) {
+			if (storeItem[idx]._iMagical)
 				iclr = COL_BLUE;
-			}
 
-			if (!smithitem[ls]._iStatFlag) {
+			if (!storeItem[idx]._iStatFlag)
 				iclr = COL_RED;
-			}
 
-			if (smithitem[ls]._iMagical) {
-				AddSText(20, l, false, smithitem[ls]._iIName, iclr, true);
+			if (storeItem[idx]._iMagical && storeItem[idx]._iIdentified) {
+				AddSText(20, l, false, storeItem[idx]._iIName, iclr, true);
+				AddSTextVal(l, storeItem[idx]._iIvalue);
 			} else {
-				AddSText(20, l, false, smithitem[ls]._iName, iclr, true);
+				AddSText(20, l, false, storeItem[idx]._iName, iclr, true);
+				AddSTextVal(l, storeItem[idx]._ivalue);
 			}
 
-			AddSTextVal(l, smithitem[ls]._iIvalue);
-			PrintStoreItem(&smithitem[ls], l + 1, iclr);
+			PrintStoreItem(&storeItem[idx], l + 1, iclr);
 			stextdown = l;
-			ls++;
 		}
+		idx++;
 	}
 
-	if (stextsel != -1 && !stext[stextsel]._ssel && stextsel != 22)
-		stextsel = stextdown;
+	stextsmax = storenumh - 4;
+	if (stextsmax < 0)
+		stextsmax = 0;
 }
 
-void S_StartSBuy()
+/**
+ * @brief Purchases an item from witch, smith or healer.
+ */
+void S_StartBuy()
 {
-	int i;
+	storenumh = 0;
+	for (int i = 0; !storeItem[i].isEmpty() && i < storeItemCount; i++)
+		storenumh++;
 
 	stextsize = true;
 	stextscrl = true;
 	stextsval = 0;
-	sprintf(tempstr, "I have these items for sale:             Your gold: %i", plr[myplr]._pGold);
+	//sprintf(tempstr, "I have these items for sale:             Your gold: %i", plr[myplr]._pGold);

```

Comment:
disabled?

Reply:
Yes...   Meant to delete this.
I've moved this into S_Smith/Witch/HealerEnter, to support Griswold's different message for premium items.

Reply:
It would be best if you could make each commit mergeable so that we could merge this gradually.